app/models/outgoing_message.rb

```diff
@@ -61,6 +61,8 @@ class OutgoingMessage < ApplicationRecord
            :inverse_of => :outgoing_message,
            :dependent => :destroy
 
+  delegate :public_body, to: :info_request, private: true, allow_nil: true
```

Comment:
> I would make this a `has_one :through` association

I think that's the wrong way round for the association, but even so, I think this is analogous:

```diff
diff --git a/app/models/outgoing_message.rb b/app/models/outgoing_message.rb
index 1aca85aaf..d74f15fe5 100644
--- a/app/models/outgoing_message.rb
+++ b/app/models/outgoing_message.rb
@@ -61,7 +61,8 @@ class OutgoingMessage < ApplicationRecord
            :inverse_of => :outgoing_message,
            :dependent => :destroy

-  delegate :public_body, to: :info_request, private: true, allow_nil: true
+  belongs_to :public_body, inverse_of: :outgoing_messages

   after_initialize :set_default_letter
   # reindex if body text is edited (e.g. by admin interface)
diff --git a/app/models/public_body.rb b/app/models/public_body.rb
index 12eca4dda..d08293a27 100644
--- a/app/models/public_body.rb
+++ b/app/models/public_body.rb
@@ -62,6 +62,11 @@ class ImportCSVDryRun < StandardError; end
   has_many :info_requests,
            -> { order('created_at desc') },
            :inverse_of => :public_body
+
+  has_many :outgoing_messages,
+           through: :info_requests,
+           inverse_of: :public_body
+
   has_many :track_things,
            -> { order('created_at desc') },
            :inverse_of => :public_body,
```

Seems like this doesn't work, at least for the specs, since it requires the models to be saved. Same when using `has_one :through`.

```
[1](#<RSpec::ExampleGroups::OutgoingMessageWhenMakingAnOutgoingMessage>)> outgoing_message.public_body
# => nil
[2](#<RSpec::ExampleGroups::OutgoingMessageWhenMakingAnOutgoingMessage>)> outgoing_message.info_request.public_body
# => #<PublicBody:0x000055f4f2346178 id: nil …>
```

Could do some work to make all this work, but I think fixing it _properly_ would be a pretty significant undertaking.

Reply:
Errm `belongs_to` is wrong there is no `public_body_id` foreign key column and it'll always return `nil` and not even try do a DB query.

When using `has_one :through` the association is properly loaded but as you say the models to be do need to be saved.

```diff
diff --git a/app/models/outgoing_message.rb b/app/models/outgoing_message.rb
index 55ca3edce..3e8dc2d09 100644
--- a/app/models/outgoing_message.rb
+++ b/app/models/outgoing_message.rb
@@ -61,7 +61,7 @@ class OutgoingMessage < ApplicationRecord
            :inverse_of => :outgoing_message,
            :dependent => :destroy
 
-  delegate :public_body, to: :info_request, private: true, allow_nil: true
+  has_one :public_body, through: :info_request
 
   after_initialize :set_default_letter
   # reindex if body text is edited (e.g. by admin interface)
diff --git a/spec/models/outgoing_message_spec.rb b/spec/models/outgoing_message_spec.rb
index 7179bdfa2..507240f87 100644
--- a/spec/models/outgoing_message_spec.rb
+++ b/spec/models/outgoing_message_spec.rb
@@ -901,8 +901,7 @@ class TestError < StandardError; end
 
     let(:message) do
       public_body = FactoryBot.build(:public_body, name: 'Example Body')
-      info_request = FactoryBot.build(:info_request, public_body: public_body)
-      FactoryBot.build(:initial_request, info_request: info_request, body: body)
+      FactoryBot.create(:initial_request, public_body: public_body, body: body)
     end
 
     let(:body) { "Dear Example Body,\n\n\r\nSome information please." }
```